<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Daniel Greene" />

<meta name="date" content="2024-02-20" />

<title>Prior phenotypic information</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>
<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Prior phenotypic information</h1>
<h4 class="author">Daniel Greene</h4>
<h4 class="date">2024-02-20</h4>



<p>The <code>sim_reg</code> function performs inference under the ‘Similarity regression’ model [1] conditional on supplied binary genotypes and ontological term sets.</p>
<p>The vignette ‘Similarity Regression - Introduction’ shows a simple application based on simulated data. This vignette demonstrates how to set the prior on <code>phi</code>, for example in order to include prior information about likely phenotype of a disease. The information is supplied to the inference procedure as a parameter called <code>term_weights</code> and should be a numeric vector of relative weights for terms included in the sample space of <code>phi</code> (by default the set of all terms present amongst the terms in <code>x</code> and their ancestors).</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true"></a><span class="kw">library</span>(ontologyIndex)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true"></a><span class="kw">library</span>(ontologySimilarity)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true"></a><span class="kw">library</span>(SimReg)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true"></a><span class="kw">data</span>(hpo)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true"></a><span class="co"># only use terms which are descended from HP:0000001</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true"></a>pa_descs &lt;-<span class="st"> </span><span class="kw">intersection_with_descendants</span>(hpo, <span class="st">&quot;HP:0000001&quot;</span>, hpo<span class="op">$</span>id)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true"></a>hpo &lt;-<span class="st"> </span><span class="kw">lapply</span>(hpo, <span class="st">&quot;[&quot;</span>, pa_descs)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true"></a><span class="kw">class</span>(hpo) &lt;-<span class="st"> &quot;ontology_index&quot;</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true"></a></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true"></a><span class="kw">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true"></a>terms &lt;-<span class="st"> </span><span class="kw">get_ancestors</span>(hpo, <span class="kw">c</span>(hpo<span class="op">$</span>id[<span class="kw">match</span>(<span class="kw">c</span>(<span class="st">&quot;Abnormality of thrombocytes&quot;</span>,<span class="st">&quot;Hearing abnormality&quot;</span>), </span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true"></a>   hpo<span class="op">$</span>name)], <span class="kw">intersection_with_descendants</span>(hpo, <span class="st">&quot;HP:0000118&quot;</span>, <span class="kw">sample</span>(hpo<span class="op">$</span>id[<span class="op">!</span>hpo<span class="op">$</span>obsolete], <span class="dt">size=</span><span class="dv">50</span>))))</span></code></pre></div>
<p>To help illustrate the ideas, we’ll consider a scenario where there is some evidence for an association between a phenotype - which in this example we’ll set as ‘Hearing abnormality’ - and a binary genotype, and where there is also a prior expectation that the ‘characteristic phenotype’ of the genotype would involve hearing abnormality. For instance, the genotype might depend on variants in a gene orthologous to one known to harbour variants associated with ‘Hearing abnormality’ in a model organism. We apply the inference procedure to the data with and without the prior and observe the effect on the inferred probability of association.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true"></a>hearing_abnormality &lt;-<span class="st"> </span>hpo<span class="op">$</span>id[<span class="kw">match</span>(<span class="st">&quot;Hearing abnormality&quot;</span>, hpo<span class="op">$</span>name)]</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true"></a>genotypes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="ot">TRUE</span>, <span class="dv">2</span>), <span class="kw">rep</span>(<span class="ot">FALSE</span>, <span class="dv">98</span>))</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true"></a><span class="co">#give all subjects 5 random terms and add &#39;hearing abnormality&#39; for those with y_i=TRUE</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true"></a>phenotypes &lt;-<span class="st"> </span><span class="kw">lapply</span>(genotypes, <span class="cf">function</span>(y_i) <span class="kw">minimal_set</span>(hpo, <span class="kw">c</span>(</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true"></a><span class="cf">if</span> (y_i) hearing_abnormality <span class="cf">else</span> <span class="kw">character</span>(<span class="dv">0</span>), <span class="kw">sample</span>(terms, <span class="dt">size=</span><span class="dv">5</span>))))</span></code></pre></div>
<p>So there are three cases with the rare variant (i.e. having y_i = TRUE) and all of them have the ‘Hearing abnormality’ HPO term.</p>
<p>An application of yields:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true"></a><span class="kw">sim_reg</span>(<span class="dt">ontology=</span>hpo, <span class="dt">x=</span>phenotypes, <span class="dt">y=</span>genotypes)</span></code></pre></div>
<pre><code>## Probability of association:  0.23 
## -------------------------------------------------------------------------------- 
##                                          Name    p
##                           Hearing abnormality 0.36
##                       Abnormal ear physiology 0.31
##                        Abnormality of the ear 0.30
##                           Champagne cork sign 0.25
##            Abnormal fetal skeletal morphology 0.23
##                     Abnormal fetal morphology 0.17
##                                 Fetal anomaly 0.16
##  Abnormality of prenatal development or birth 0.11
##     Abnormality of the musculoskeletal system 0.00
##            Abnormal axial skeleton morphology 0.00
## --------------------------------------------------------------------------------</code></pre>
<p>We now consider constructing the <code>term_weights</code> parameter to capture our knowledge about the gene from the model organism. We can either explicitly create the <code>term_weights</code> vector of prior weights, i.e. by assigning higher weights to terms which involve some kind of hearing problem. For example, we could set the prior weight of all terms which have the word ‘hearing’ in to ten times that of terms which don’t.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true"></a>term_weights &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">grepl</span>(<span class="dt">x=</span>hpo<span class="op">$</span>name, <span class="dt">ignore=</span><span class="ot">TRUE</span>, <span class="dt">pattern=</span><span class="st">&quot;hearing&quot;</span>), <span class="dv">10</span>, <span class="dv">1</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true"></a><span class="kw">names</span>(term_weights) &lt;-<span class="st"> </span>hpo<span class="op">$</span>id</span></code></pre></div>
<p>Note: one must set the <code>names</code> of the <code>term_weights</code> vector, as <code>sim_reg</code> will use it.</p>
<p>If the prior knowledge of the phenotype/phenotype of the model organism has been ontologically encoded (for example, it may be available as MPO terms from the Mouse Genome Informatics (MGI) website, <a href="https://www.informatics.jax.org/" class="uri">https://www.informatics.jax.org/</a>, [2]), another option is to use a phenotypic similarity function to obtain the numeric vector of weights for inclusion of terms in <code>phi</code> [1]. This may be more convenient, particularly when dealing with large numbers of genes. In the SimReg paper [1], the vector is set using the Resnik-based [3] similarities of terms to the terms in the ‘literature phenotype’. In order to calculate the similarities based on Resnik’s similarity measure, we must first compute an ‘information content’ for the terms, equal to the negative log frequency. The frequencies can be calculated with respect to different collections of phenotypes. Here, we will calculate it with respect to the frequencies of terms within our collection, <code>phenotypes</code>, by calling the function exported by the <code>ontologyIndex</code> package, <code>get_term_info_content</code>. Note, it could also be calculated with respect to the frequency of the term amongst the ontological annotation of OMIM diseases (available from the HPO website, <a href="http://human-phenotype-ontology.github.io" class="uri">http://human-phenotype-ontology.github.io</a> [4]). The function <code>get_term_set_to_term_sims</code> in the package <code>ontologySimilarity</code> can then be used to calculate the similarities between the terms in the sample space of <code>phi</code> and the <code>literature_phenotype</code>. It calculates a matrix of similarities between the individual terms in the literature phenotype and terms in the sample space. Let’s say the phenotype of the model organism in our example contains abnormalities of the thrombocytes and hearing abnormality.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true"></a>thrombocytes &lt;-<span class="st"> </span>hpo<span class="op">$</span>id[<span class="kw">match</span>(<span class="st">&quot;Abnormality of thrombocytes&quot;</span>, hpo<span class="op">$</span>name)]</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true"></a>literature_phenotype &lt;-<span class="st"> </span><span class="kw">c</span>(hearing_abnormality, thrombocytes)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true"></a>info &lt;-<span class="st"> </span><span class="kw">get_term_info_content</span>(hpo, phenotypes)</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true"></a>term_weights_resnik &lt;-<span class="st"> </span><span class="kw">apply</span>(<span class="kw">get_term_set_to_term_sims</span>(</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true"></a>    <span class="dt">ontology=</span>hpo, <span class="dt">information_content=</span>info, <span class="dt">terms=</span><span class="kw">names</span>(info),</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true"></a>    <span class="dt">term_sim_method=</span><span class="st">&quot;resnik&quot;</span>, <span class="dt">term_sets=</span><span class="kw">list</span>(literature_phenotype)), <span class="dv">2</span>, mean)</span></code></pre></div>
<p>This can then be passed to <code>sim_reg</code> through the <code>term_weights</code> parameter.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true"></a><span class="kw">sim_reg</span>(</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true"></a>    <span class="dt">ontology=</span>hpo,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true"></a>    <span class="dt">x=</span>phenotypes,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true"></a>    <span class="dt">y=</span>genotypes,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true"></a>    <span class="dt">term_weights=</span>term_weights_resnik</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true"></a>)</span></code></pre></div>
<pre><code>## Probability of association:  0.39 
## -------------------------------------------------------------------------------- 
##                                          Name    p
##                           Hearing abnormality 0.40
##                       Abnormal ear physiology 0.31
##                        Abnormality of the ear 0.28
##                           Champagne cork sign 0.22
##            Abnormal fetal skeletal morphology 0.21
##                     Abnormal fetal morphology 0.15
##                                 Fetal anomaly 0.15
##  Abnormality of prenatal development or birth 0.10
##     Abnormality of the musculoskeletal system 0.00
##            Abnormality of the skeletal system 0.00
## --------------------------------------------------------------------------------</code></pre>
<p>Note that including the <code>term_weights</code> parameter has increased the mean posterior value of <code>gamma</code>.</p>
<p>Often the binary genotype relates to a particular gene, and for many genes ontologically encoded phenotypes are available either in the form of HPO encoded OMIM annotations [4] or MPO annotations [2]. For a given set of subjects with HPO-coded phenotypes, it may be useful to apply the inference gene-by-gene, taking the binary genotype <code>y</code> to indicate the presence of a rare variant in each particular gene for each case. Thus, we may wish to systematically create informative prior distributions for <code>phi</code> for all genes. This can be done by downloading the file called ‘ALL_SOURCES_TYPICAL_FEATURES_genes_to_phenotype.txt’ from the HPO website, and running the following code yielding a list of term sets (i.e. character vectors of HPO term IDs).</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true"></a>annotation_df &lt;-<span class="st"> </span><span class="kw">read.table</span>(<span class="dt">header=</span><span class="ot">FALSE</span>, <span class="dt">skip=</span><span class="dv">1</span>, <span class="dt">sep=</span><span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, </span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true"></a>    <span class="dt">file=</span><span class="st">&quot;ALL_SOURCES_TYPICAL_FEATURES_genes_to_phenotype.txt&quot;</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>, <span class="dt">quote=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true"></a>hpo_by_gene &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">split</span>(<span class="dt">f=</span>annotation_df[,<span class="dv">2</span>], <span class="dt">x=</span>annotation_df[,<span class="dv">4</span>]), </span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true"></a>    <span class="cf">function</span>(trms) <span class="kw">minimal_set</span>(hpo, <span class="kw">intersect</span>(trms, hpo<span class="op">$</span>id)))</span></code></pre></div>
<div id="references" class="section level2">
<h2>References</h2>
<ol style="list-style-type: decimal">
<li>D. Greene, NIHR BioResource, S. Richardson, E. Turro, Phenotype similarity regression for identifying the genetic determinants of rare diseases, The American Journal of Human Genetics 98, 1-10, March 3, 2016.</li>
<li>Eppig JT, Blake JA, Bult CJ, Kadin JA, Richardson JE;; The Mouse Genome Database Group. 2015. The Mouse Genome Database (MGD): facilitating mouse as a model for human biology and disease. Nucleic Acids Res. 2015 Jan 28;43(Database issue):D726-36.</li>
<li>Philip Resnik (1995). Chris S. Mellish (Ed.), ed. Using information content to evaluate semantic similarity in a taxonomy. Proceedings of the 14th international joint conference on Artificial intelligence (IJCAI’95) (Morgan Kaufmann Publishers Inc., San Francisco, CA, USA) 1: 448-453.</li>
<li>Sebastian Kohler, Sandra C Doelken, Christopher J. Mungall, Sebastian Bauer, Helen V. Firth, et al. The Human Phenotype Ontology project: linking molecular biology and disease through phenotype data Nucl. Acids Res. (1 January 2014) 42 (D1): D966-D974</li>
</ol>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
